workalendar/core.py

```diff
@@ -1010,16 +1018,18 @@ def __init__(self, **kwargs):
     def get_fixed_holidays(self, year):
         days = super().get_fixed_holidays(year)
         if self.include_new_years_day:
-            days.insert(
-                0, (date(year, 1, 1), "New year")
-            )
-
+            days.insert(0, (date(year, 1, 1), "New year"))
+        if self.include_new_years_eve:
+            days.append((date(year, 12, 31), "New Year's eve"))
         if self.include_labour_day:
-            days.append(
-                (date(year, 5, 1), self.labour_day_label)
-            )
+            days.append((date(year, 5, 1), self.labour_day_label))
         return days
 
+        if self.include_islamic_new_year:
```

Comment:
I think there's a mistake here: your line 1028 and the rest are located right after the "return" statement at line 1026... they'll never be executed, I'm afraid.

And while we're at it, if those non-working days are related to the Islamic holidays, I'd prefer them to be in the Islamic mixin class, rather than the core one.

Reply:
This was a wrong copy pasted code. Removed line 1028 onwards